src/04 - Web services.md

```diff
@@ -103,58 +116,230 @@ Content-Type: application/json
 
-## **bsm-registry**
+## **bsm-registry/group/{groupId}**
 
-Gets the definitions of all business-specific metadata fields available in the system, the definitions of a group of fields, or the definition of a single field.
+### create and update
+
+Creates or updates a business-specific metadata group definition.
 
 **Request**
 
 ``` text
-GET /rms/api/public/noark5/v1/bsm-registry?groupId=GROUP_ID&fieldId=FIELD_ID HTTP/1.1
+PUT /rms/api/public/noark5/v1/bsm-registry/group/{groupId} HTTP/1.1
 Authorization: Bearer ACCESS_TOKEN
 Content-Type: application/json
+
+{
+  "groupName": string,
+  "groupDescription": string
+}
 ```
 
 ###### Details
 
-- **groupId** (optional)
-  - the group ID of the field definitions to be fetched
-- **fieldId** (optional)
-  - the field ID of the field definition to be fetched
+- **groupId**
+  - group ID of the group definition to be created or updated (a string comprised of lower-case letters, digits, and dashes that must begin with a letter and end with a letter or a digit)
+- **groupName**
+  - group name
+- **groupDescription**
+  - group description
 
-If **groupId** and **fieldId** are both not specified, the response will contain all business-specific metadata field definitions.
+**Response**
+
+``` text
+Content-type: application/json
+
+{
+  "groupId": string,
+  "groupName": string,
+  "groupDescription": string
+}
+```
+
+###### Details
+
+- **groupId**
+  - system unique group ID of the created or updated group definition (a string comprised of lower-case letters, digits, and dashes that must begin with a letter and end with a letter or a digit)
+- **groupName**
+  - group name
+- **groupDescription**
+  - group description
+
+### delete
+
+Deletes a business-specific metadata group definition.
+
+**Request**
+
+``` text
+DELETE /rms/api/public/noark5/v1/bsm-registry/group/{groupId} HTTP/1.1
+Authorization: Bearer ACCESS_TOKEN
+```
+
+###### Details
+
+- **groupId**
+  - group ID of the group definition to be deleted (a string comprised of lower-case letters, digits, and dashes that must begin with a letter and end with a letter or a digit)
+
+**Response**
+
+No content will be returned upon successful completion.
+
+## **bsm-registry/group/{groupId}/field/{fieldId}**
+
+### create and update
+
+Creates or updates a business-specific metadata field definition.
+
+**Request**
+
+``` text
+PUT /rms/api/public/noark5/v1/bsm-registry/group/{groupId}/field/{fieldId} HTTP/1.1
+Authorization: Bearer ACCESS_TOKEN
+Content-Type: application/json
+
+{
+  "fieldName": string,
+  "fieldDescription": string,
+  "fieldType": string|long|double,
+  "fieldValues": [string|long|double]
+}
+```
+
+###### Details
+
+- **groupId**
+  - group ID of the group definition in which to create or update the field definition
+- **fieldId**
+  - field ID of the field definition to be created or updated
+- **fieldName**
+  - field name
+- **fieldDescription**
+  - group description
```

Comment:
I think this should read `field description`.

Reply:
Fixed.